apps/backoffice-v2/src/pages/Entities/components/CaseGeneration/context/case-generation-context/hooks/useCaseGenerationState/useCaseGenerationState.ts

```diff
@@ -0,0 +1,39 @@
+import { initialCaseGenerationContext } from '@/pages/Entities/components/CaseGeneration/context/case-generation-context/case-generation.context';
+import {
+  CaseGenerationApi,
+  CaseGenerationContext,
+} from '@/pages/Entities/components/CaseGeneration/context/case-generation-context/types';
+import { useCallback, useMemo, useState } from 'react';
+
+export const useCaseGenerationAPI = () => {
```

Comment:
My initial thought was this hook makes HTTP requests. Is using our existing naming for hooks for components business logic applicable here?
```suggestion
export const useCaseGenerationLogic = () => {
```